### DISCUSSION TITLE
[Discussion] Integer overflow

### DISCUSSION THREAD

**Discussion by posidoni** [bezlant/s21_bigint#33] [Ideas]
In functions that convert decimal to sth else we should think about overflow. See below NPZ functions to check these cases.

```
Function: int mpz_fits_ulong_p (const mpz_t op)
Function: int mpz_fits_slong_p (const mpz_t op)
Function: int mpz_fits_uint_p (const mpz_t op)
Function: int mpz_fits_sint_p (const mpz_t op)
Function: int mpz_fits_ushort_p (const mpz_t op)
Function: int mpz_fits_sshort_p (const mpz_t op)
Return non-zero iff the value of op fits in an unsigned long int, signed long int, unsigned int, signed int, unsigned short int, or signed short int, respectively. Otherwise, return zero.

Macro: int mpz_odd_p (const mpz_t op)
Macro: int mpz_even_p (const mpz_t op)


```

**Comment by bezlant** [comment]
I've set up a check in decimal to int for all the overflows.
I'm considering if we need one for FLOAT, seems like floats can be larger than our 96bit decimal. 
<img width="634" alt="float sizes" src="https://user-images.githubusercontent.com/89563512/172309946-327a4c35-33b2-4ae0-a65e-1622116f3927.png">

**Comment by posidoni** [comment]
I will check this issue one more time today & will close it if everything is okay.
